homeassistant/components/rointe/climate.py

```diff
@@ -0,0 +1,275 @@
+
+    @property
+    def max_temp(self) -> float:
+        """Max selectable temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_max_temp
+
+        return RADIATOR_TEMP_MAX
+
+    @property
+    def min_temp(self) -> float:
+        """Minimum selectable temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_min_temp
+
+        return RADIATOR_TEMP_MIN
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Max selectable target temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_max_temp
+
+        return RADIATOR_TEMP_MAX
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Minimum selectable target temperature."""
+        if self._radiator.user_mode_supported and self._radiator.user_mode:
+            return self._radiator.um_min_temp
+
+        return RADIATOR_TEMP_MIN
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Flag supported features."""
+        return (
+            ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+        )
+
+    @property
+    def target_temperature_step(self) -> float | None:
+        """Temperature step."""
+        return RADIATOR_TEMP_STEP
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return hvac modes available."""
+        return AVAILABLE_HVAC_MODES
+
+    @property
+    def preset_modes(self) -> list[str]:
+        """Return the available preset modes."""
+        return AVAILABLE_PRESETS
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current HVAC mode."""
+        if not self._radiator.power:
+            return HVACMode.OFF
+
+        if self._radiator.mode == RADIATOR_MODE_AUTO:
+            return HVACMode.AUTO
+
+        return HVACMode.HEAT
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current HVAC action."""
+
+        # Special mode for AUTO mode and waiting for schedule to activate.
+        if (
+            self._radiator.mode == RADIATOR_MODE_AUTO
+            and self._radiator.preset == HVACMode.OFF
+        ):
+            return HVACAction.IDLE
+
+        # Forced to off, either on Manual or Auto mode.
+        if not self._radiator.power:
+            return HVACAction.OFF
+
+        # Otherwise, it's heating.
+        return HVACAction.HEATING
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Convert the device's preset to HA preset modes."""
+
+        if self._radiator.preset == RADIATOR_PRESET_ECO:
+            return PRESET_ECO
+        if self._radiator.preset == RADIATOR_PRESET_COMFORT:
+            return PRESET_COMFORT
+        if self._radiator.preset == RADIATOR_PRESET_ICE:
+            return PRESET_ROINTE_ICE
```

Comment:
Instead of all these return statements, I would make a dict (personally, a class variable) that maps radiator presets to presets.